psimol/main.py

```diff
@@ -452,7 +502,114 @@ def from_cif(cls, file_path: str) -> Molecule:
+                            return None
+                
+                # assemble the body of the loop
+                if in_atom_site_body:
+                    # a loop may be terminated by:
+                    # - a line containing a single '#' and otherwise empty
+                    # - a new non-looped data item
+                    # - an empty line
+                    # - start of a new loop
+                    if line[0] == '#' or line[0] == '_' or line == '' or line[:5].lower() == 'loop_':
+                        break
+                    atom_site_body += line
+
+        # phase 3: parse loop
+        # .cif file specification states a length limit for lines.
+        # To satisfy this limit, overlong records in loops may be broken into multiple lines.
+        # This caveat essentially makes the newline no different from any other whitespace,
+        # which necessitates treating the body of the loop in a flat manner.
+
+        # guaranteed to mangle any quoted items containing whitespace
+        atom_site_body = atom_site_body.split()
+        atoms = []
+        symbol_idx = atom_site_header['type_symbol']
+        x_idx = atom_site_header['cartn_x']
+        y_idx = atom_site_header['cartn_y']
+        z_idx = atom_site_header['cartn_z']
+        nrow = len(atom_site_body) // ncol
+        # check that the number of items is as would be expected based on the number of rows and columns
+        if nrow*ncol != len(atom_site_body):
+            # if it isn't, a possible cause is that a quoted item got mangled
+            logging.error('Failed to parse .cif file, possibly incompatible')
+            # since this would mess up all the remaining items, the parser gives up
+            return None
+
+        for i in range(nrow):
+
+            offset = i*ncol
+
+            # extract atom's items from `body'
+            symbol: str = atom_site_body[offset+symbol_idx]
+            x: str = atom_site_body[offset+x_idx]
+            y: str = atom_site_body[offset+y_idx]
+            z: str = atom_site_body[offset+z_idx]
+
+            # '.' and '?' signify unapplicable and missing values
+            # the parser can't cope with missing coord or element info and gives up
+            if any(( item == '?' or item == '.' for item in (symbol, x, y, z) )):
+                logging.error('.cif file is missing an atom\'s element or coordinate')
+                return None
+
+            # numeric values in .cif files may be appended with an uncertainty in parenthesis
+            # the parser ignores this
+            for coord in x, y, z:
+                parenthesis = coord.rfind('(')
+                if parenthesis != -1:
+                    coord = coord[:parenthesis]
+
+            atom = Atom(symbol, name=i+1, x=float(x), y=float(y), z=float(z))
+            atoms.append(atom)
+
+        # get conjectural name
+        name = file_path.split('/')[-1]
```

Comment:
```suggestion
        name = os.path.basename(file_path)
```
We can add `import os`, it would be more OS independent.